substrate/primitives/runtime/src/traits/mod.rs

```diff
@@ -1474,8 +1551,57 @@ impl Dispatchable for () {
 	}
 }
 
+/// Dispatchable impl containing an arbitrary value which panics if it actually is dispatched.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct FakeDispatchable<Inner>(pub Inner);
+impl<Inner> From<Inner> for FakeDispatchable<Inner> {
+	fn from(inner: Inner) -> Self {
+		Self(inner)
+	}
+}
+impl<Inner> FakeDispatchable<Inner> {
+	/// Take `self` and return the underlying inner value.
+	pub fn deconstruct(self) -> Inner {
+		self.0
+	}
+}
+impl<Inner> AsRef<Inner> for FakeDispatchable<Inner> {
+	fn as_ref(&self) -> &Inner {
+		&self.0
+	}
+}
+
+impl<Inner> Dispatchable for FakeDispatchable<Inner> {
+	type RuntimeOrigin = ();
+	type Config = ();
+	type Info = ();
+	type PostInfo = ();
+	fn dispatch(
+		self,
+		_origin: Self::RuntimeOrigin,
+	) -> crate::DispatchResultWithInfo<Self::PostInfo> {
+		panic!("This implementation should not be used for actual dispatch.");
+	}
+}
+
+/// Runtime Origin which includes a System Origin variant whose `AccountId` is the parameter.
+pub trait AsSystemOriginSigner<AccountId> {
+	/// Extract a reference of the inner value of the System `Origin::Signed` variant, if self has
+	/// that variant.
+	fn as_system_origin_signer(&self) -> Option<&AccountId>;
+}
+
+/// Interface to differentiate between authorized and unauthorized Runtime Origins. Generally, an
+/// authorized origin is either an `Origin::Signed` System origin or a custom origin auhtorized in a
+/// `TransactionExtension`.
+pub trait AsAuthorizedOrigin {
+	/// Returns `true` if the origin is a System `Origin::None` variant, `false` otherwise.
```

Comment:
This doc is too specific for sp-runtime, maybe we can say 
```suggestion
	/// Whether the origin is authorized to include the transaction in a block.
	///
	/// In FRAME returns `false` if the origin is a System `Origin::None` variant, `true` otherwise, meaning 
	/// only signed or custom origin resulting from the transaction extension pipeline are authorized.
```

Reply:
Done.